Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -846,6 +847,14 @@ data PackageCheck =
      | PackageDistInexcusable { explanation :: CheckExplanation }
   deriving (Eq, Ord)
 
+-- | Would Hackage refuse a package because of this error?
+isHackageDistError :: PackageCheck -> Bool
+isHackageDistError (PackageBuildImpossible {}) = True
```

Comment:
I usually use lambda-case in such cases, so that you don't have to repeat the function name numerous times. But up to you...